Sources/BraintreeCoreSwift/BTHTTP.swift

```diff
@@ -69,7 +74,13 @@ import Security
     /// - Parameter clientToken: The client token
     @objc(initWithClientToken:error:)
     public convenience init(clientToken: BTClientToken) throws {
-        guard let clientApiURL = clientToken.json["clientApiUrl"].asURL() else {
+        let url: URL
+
+        if clientToken.json["clientApiUrl"].asURL() != nil, let clientApiURL = clientToken.json["clientApiUrl"].asURL() {
+            url = clientApiURL
+        } else if clientToken.json["configUrl"].asURL() != nil, let configURL = clientToken.json["configUrl"].asURL() {
+            url = configURL
```

Comment:
Why are we adding this logic here? Not seeing where that comes from in the original BTHTTP.m?

Reply:
Now that the URL is required as part of the init here: `self.init(url: url, authorizationFingerprint: clientToken.authorizationFingerprint)` we need to get this URL from the client token. In v1 and v2 client tokens the URL is in the JSON at `clientApiUrl`. In v3 client tokens the URL is in the JSON at `configUrl`. 

We did not catch this when converting `BTHTTP` because we just converted the file but hadn't updated anything else to use the Swift version of `BTHTTP`. Once we updated the SDK to use all of the Swift classes, the integration tests using `SANDBOX_CLIENT_TOKEN_VERSION_3` began to fail. Decoding the client tokens in `IntegrationTestsConstants` showed the change in the URL between v1/v2 and v3 client tokens.

Reply:
The v1/v2 client tokens contain both a `clientApiUrl` and `configUrl` while v3 just contains a `configUrl`. So it's possible that we can just use the `configUrl` and remove the reference to `clientApiUrl`?

Reply:
Got it - thank you for the thorough explanation! Yeh, it would be interesting to know which should take priority. Did we check what Android and Web are doing?